docs/DeveloperGuide.md

```diff
@@ -626,7 +626,161 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+We will check the existence of the `IC_NUMBER` first during the execution of the `edit` command. 
+
+**Examples**
+
+`edit S1234567P` should return the error message *the IC_NUMBER provided does not exist.*
+
+### Allowing for multiple phone numbers
+
+**Potential Issues With Current Feature**
+
+Currently, our `phone` number field only allows for one number. However, since out product is targeted at doctors working at the GP clinics with their contacts being patients, it 
+is only natural for the `phone` number field to allow for multiple numbers. Reason being that a patient might have a caregiver or is required to contact his/her 
+family members. Therefore, restricting the `phone` to only one is very restrictive and could bring about inconvenience.
+
+**Proposed Enhancements**
+
+The team will thus allow for the addition of multiple numbers to be added under the `phone` field through the `add` command.
+
+**Examples**
+
+- `add n\John Doe p\12345678 e\johndoe@mail.com i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25` will cause the `phone` number `12345678` to be added for user `T0123456A`
+- `add n\John Doe p\12345678, 09876544, 99999999 e\johndoe@mail.com i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25` will cause the `phone` number `12345678, 09876544, 99999999` to be added for user `T0123456A`
+
+### Allowing for derivation of age through Date-Of-Birth
+
+**Potential Issues With Current Feature**
+
+Existing features require the users to manually ass the age of contacts. For example, `add n\John Doe p\12345678 e\johndoe@mail.com i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25`.
+However, it if very cumbersome for users to edit the age of their contact individually through the `edit` command.
+
+**Proposed Enhancements**
+
+In order to allow to save the time of our users, we will adding in a new field called `DOB` which Date of Birth in the form of `DD/MM/YYYY`. 
+`age` of the contacts will thus be derived from their `DOB` with referenced to the current year which we are in. Hence, streamlining the contact management process.
+
+**Examples**
+
+Given that we are in the year 2024, the patient with the `DOB` of 
+
+- `10/10/2003` will be at the age of 21 
+- `12/12/1990` will be at the age of 34
+
+### Further enhancements for `addnote` command
+
+**Potential Issues With Current Feature**
+
+There were many constructive feedbacks given regarding the `addnote` command.
+
+Firstly, our current `addnote` command only allows user to add a new note, which will be appended to the previous one. However,
+there might be many instances in which the users need to edit or delete specific note. Disallowing such commands will make contact quite frustrating at times
+
+Secondly, our patient profile card has a notes section which cannot be hidden and will appear in the UI. Users thus might not want to see the notes section in the patient's profile card as it might be very long and distracting, making it unpleasant to the eyes.
```

Comment:
Secondly, each of the contacts in the patient list panel has a notes section which cannot be hidden and will appear in the UI. Users might not want to see all the notes for each contact as it might be very long and distracting.